internal/host/validator.go

```diff
@@ -658,4 +660,172 @@ func (v *validator) printSufficientOrUnknownInstallationDiskSpeed(c *validationC
 	default:
 		return fmt.Sprintf("Unexpected status %s", status.String())
 	}
+
+}
+
+func (v *validator) hasNetworkLatencyRequirementForRole(c *validationContext) ValidationStatus {
+
+	if len(c.host.Connectivity) == 0 {
+		return ValidationPending
+	}
+	latency, err := v.validateNetworkLatencyForRole(c.host, c.clusterHostRequirements, c.cluster.Hosts)
+	if err != nil || len(latency) > 0 {
+		return ValidationFailure
+	}
+	return ValidationSuccess
+}
+
+func getHostnameAndRoleByIP(ip string, hosts []*models.Host) (string, models.HostRole, error) {
+
+	for _, h := range hosts {
+		if h.Inventory != "" {
+			var inv models.Inventory
+			err := json.Unmarshal([]byte(h.Inventory), &inv)
+			if err != nil {
+				return "", "", fmt.Errorf("unable to unmarshall cluster inventory for host %s", h.RequestedHostname)
+			}
+			if strings.Count(ip, ":") < 2 {
```

Comment:
When does an IPv4 address contain any colons?

Reply:
When it is provided with the port. It's a safeguard but I can remove it if the general consensus is that this particular use case is never going to occur.

Reply:
Inventory does not contain ports

Reply:
I'll remove this then :)